compiler/plugins/target/ROCM/builtins/ukernel/CMakeLists.txt

```diff
@@ -45,30 +29,27 @@ function(iree_rocm_bitcode_library)
   endif()
 
   set(_ROCM_ARCH "${_RULE_ROCM_ARCH}")
-  set(OPT_FLAG "-O0")
-  if(_ROCM_ARCH MATCHES "GFX9")
-    set(OPT_FLAG "-O3")
-  endif()
   set(_COPTS
-    "-x" "hip"
-
-    # Compile only the device code for the target architecture.
-    "--offload-device-only"
-    "--offload-arch=${_ROCM_ARCH}"
+    "-x" "c"
+    "-Xclang" "-finclude-default-header"
 
-    # Suppress warnings about about ROCM version (we mostly don't care).
-    "-D_ALLOW_COMPILER_AND_STL_VERSION_MISMATCH"
+    "-std=c23"
+    "-nogpulib"
+    "-fno-short-wchar"
 
-    # Use the ROCM specified by the IREE cmake variable (instead of guessing
-    # or failing if ROCM is not on the user's path).
-    "--rocm-path=${IREE_ROCM_PATH}"
+    # Target architecture/machine.
+    "-target" "amdgcn-amd-amdhsa"
+    "-march=${_ROCM_ARCH}"
```

Comment:
Do we want the features as well here? (`:sramecc+:xnack-` is pretty common on gfx9s). I think there's a few ways to get the full string, `rocm_agent_enumerator -name` is the one that comes to bind.

It's probably not _that_ big a deal, but seemed worth mentioning

Reply:
Sounds good, this is just my naive initial port using the flags from https://github.com/iree-org/iree/blob/users/benvanik/amdgpu/build_tools/cmake/iree_amdgpu_library.cmake#L49 . I hadn't heard of AMDGPU target feature sets yet, but that sounds like something we might want in the future if it's not already implied by the `-march=gfx942` flags we are passing here?

Reply:
Those are optional flags I think. BTW, kind of rusty in LLVM, but since we are only using this to generate the LLVMIR which we will then link and inline to the dispatch, wouldn't we be getting/using these flags from ROCMTarget as seen here https://github.com/iree-org/iree/blob/540cebfa07e9cbb5e421c20da961a934ea3cb166/compiler/plugins/target/ROCM/ROCMTarget.cpp#L91-L94 anyways?

Reply:
The late post-linking passes would indeed run with those features. But enabling features also controls the semantics of the source code (e.g., if my CPU experience is any guide here, they might make certain builtins available, and make corresponding preprocessor tokens defined). So it may matter to have these features enabled at the time of the compilation of ukernels from C source.  Anyway, we are already passing `-march=` flags, such as `-march=gfx942`, which, in my understanding, suffice to enable the corresponding features. For example, https://llvm.org/docs/AMDGPUUsage.html says that `gfx942` implies `sramecc,tgsplit,xnack,kernarg,preload`. So I amn't sure when we would ever need to deal with individual features like this.

Reply:
The two ones I mention (`sramecc` and `xnack`) only come up in backend instruction selection stuff. `sramecc+` means there's error-correcting SRAM (which makes certain instructions unsafe) and `xnack-` has to do with ... some hardware mode I forget the details of, It's debugging related, but I know that leaving off the `xnack-` can have, at the very least, perf implications

But yeah, if you're just going to LLVM bitcode, shouldn't be needed.

Reply:
Can you confirm that these are implies by merely passing `gfx942` as the target architecture (both in ukernel compilation, and in IREE codegen) ?